modules/page-memory/src/test/java/org/apache/ignite/internal/pagememory/persistence/checkpoint/CheckpointerTest.java

```diff
@@ -316,6 +357,50 @@ void testDoCheckpoint() throws Exception {
         assertEquals(checkpointer.currentProgress().currentCheckpointPagesCount(), 3);
     }
 
+    @Test
+    void testNextCheckpointInterval() {
+        AtomicLong checkpointFrequency = new AtomicLong();
+        AtomicInteger checkpointFrequencyDeviation = new AtomicInteger();
+
+        Checkpointer checkpointer = new Checkpointer(
+                log,
+                "test",
+                null,
+                null,
+                mock(CheckpointWorkflow.class),
+                mock(CheckpointPagesWriterFactory.class),
+                1,
+                checkpointFrequency::get,
+                checkpointFrequencyDeviation::get
+        );
+
+        // Checks case 0 deviation.
+
+        checkpointFrequencyDeviation.set(0);
+
+        checkpointFrequency.set(1_000);
+        assertEquals(1_000, checkpointer.nextCheckpointInterval());
+
+        checkpointFrequency.set(2_000);
+        assertEquals(2_000, checkpointer.nextCheckpointInterval());
+
+        // Checks for non-zero deviation.
+
+        checkpointFrequencyDeviation.set(10);
+
+        assertThat(
+                checkpointer.nextCheckpointInterval(),
+                allOf(greaterThanOrEqualTo(1_900L), lessThanOrEqualTo(2_100L))
+        );
+
+        checkpointFrequencyDeviation.set(200);
```

Comment:
I suppose that we should have an upper limit for this value, no one will ever set it to 200, it's insane :)

Reply:
Fix it.